elasticai/creator/vhdl/language.py

```diff
@@ -0,0 +1,245 @@
+"""
+The module contains classes and functions for generating vhdl code.
+We provide code generators for the subset of vhdl that we need for implementing
+our neural network accelerators and test benches. We stick closely to the vhdl
+formal grammar with our class names.
+"""
+from abc import abstractmethod
+from enum import Enum
+from itertools import filterfalse, chain
+from typing import (
+    Callable,
+    Iterable,
+    Union,
+    NewType,
+    Literal,
+    overload,
+    Optional,
+)
+from collections import Sequence
+
+Identifier = str
+Code = Iterable[str]
+CodeGenerator = Callable[[], Code]
+
+
+class Keywords(Enum):
```

Comment:
I think the idea is nice but I wouldn't say I get everything in this implementation. A documentation would be nice, also about the usage

Reply:
allright, i'll try to improve it

Reply:
hmmm... to be honest i don't really know what to add here. The implementation tries to stick very close to the vhdl formal grammar, so i don't think it makes a lot of sense without knowing or referring to the grammar.

Reply:
Not for now, I will look at it tomorrow

Reply:
we could also just talk about it in a quick meeting at some point

Reply:
rresolve with reference to #20